diff --git a/sample/build.gradle b/sample/build.gradle
index b7b5bc3..a4b02c7 100644
--- a/sample/build.gradle
+++ b/sample/build.gradle
@@ -35,14 +35,6 @@ android {
 }
 
 dependencies {
-
-    implementation 'androidx.core:core-ktx:1.7.0'
-    implementation 'androidx.appcompat:appcompat:1.4.1'
-    implementation 'com.google.android.material:material:1.5.0'
-    implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
-    implementation 'androidx.navigation:navigation-fragment-ktx:2.3.5'
-    implementation 'androidx.navigation:navigation-ui-ktx:2.3.5'
-    testImplementation 'junit:junit:4.13.2'
-    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+    implementation libs.bundles.androidx
+    implementation libs.google.android.material
 }
\ No newline at end of file
diff --git a/settings.gradle b/settings.gradle
index c84afdf..66c3330 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -15,3 +15,4 @@ dependencyResolutionManagement {
 rootProject.name = "Library"
 include ':library'
 enableFeaturePreview('VERSION_CATALOGS')
+include ':sample'
